src/components/MovieList/MovieCard.jsx

```diff
@@ -1,33 +1,89 @@
-import "../MovieList/MovieCard.css"
-import MovieModal from "../MovieModal/MovieModal";
-import { movieDetails } from "../../services/moviesAPI";
-import {useState} from 'react'
-import { FaEye, FaStar} from 'react-icons/fa'
+import "../MovieList/MovieCard.css";
+import { useState, useEffect } from "react";
+import { FaEyeSlash, FaRegStar, FaStar, FaCheckCircle } from "react-icons/fa";
+import { favoritesUtils, watchedUtils } from "../../utils/localStorageUtils";
 
+const IMG_BASE_URL = "https://image.tmdb.org/t/p/w500";
 
-const IMG_BASE_URL = 'https://image.tmdb.org/t/p/w500';
+export default function MovieCard({
+  item,
+  onClick,
+  isFavorited = false,
+  unFavorited,
+  isWatched = false,
+  unWatched,
+}) {
+  const [isSeen, setIsSeen] = useState(isWatched);
+  const [isFavorite, setIsFavorite] = useState(isFavorited);
 
+  useEffect(() => {
+    if (!isFavorited) setIsFavorite(favoritesUtils.exist(item.id));
+    if (!isWatched) setIsSeen(watchedUtils.exist(item.id));
+  }, [item.id]);
 
+  const toggleWatched = (e) => {
+    e.stopPropagation();
+    if (isSeen) {
+      console.log("toggleWatched item:", item);
+      console.log("toggleWatched item.id:", item.id)
+      watchedUtils.remove(item.id);
+      unWatched?.();
+    } else {
+      const movie = {
+        id: item.id,
+        title: item.title,
+        poster_path: item.poster_path,
+        vote_average: item.vote_average,
+      };
+      watchedUtils.save(movie);
+    }
+    setIsSeen(!isSeen);
+  };
 
-export default function MovieCard({item, onClick}) {
-    return (
-        <>
-        <div className="movie-card" onClick={onClick}>
-            <img 
-                className="poster" 
-                src={`${IMG_BASE_URL}${item.poster_path}`} 
-                alt={"Cover of " + item.title}
-            />
+  const toggleFavorite = (e) => {
+    e.stopPropagation();
+    if (isFavorite) {
+      console.log("toggleFavorite item:", item);
+      console.log("toggleFavorite item.id:", item.id)
+      favoritesUtils.remove(item.id);
+      unFavorited?.();
```

Comment:
```suggestion
      unFavorited();
```
with unFavorited having a default with a previous comment, this optional chaining is not needed.